src/Model.elm

```diff
@@ -179,17 +179,22 @@ update msg model =
                     ( { model | error = Just (toString err) }, Cmd.none )
 
         DeleteRecord recordId ->
-            ( model, deleteRecord recordId )
+            ( model, deleteRecord model recordId )
 
         DeleteRecordResponse response ->
             case response of
                 Ok data ->
-                    ( { model
-                        | records = removeRecordFromList data model.records
-                        , error = Nothing
-                      }
-                    , fetchRecords model
-                    )
+                    case (decodeValue decodeRecord data) of
+                        Err msg ->
+                            ( { model | error = Just msg }, Cmd.none )
+
+                        Ok record ->
+                            ( { model
+                                | records = removeRecordFromList record model.records
+                                , error = Nothing
+                              }
+                            , fetchRecords model
+                            )
 
                 Err err ->
                     ( { model | error = Just (toString err) }, Cmd.none )
```

Comment:
I think we could indeed something a bit more "clever" in this case, but I'm not going to do anything for now, as this is "just" thedemo/experiment code, and not related to the API/lib itself. I might revisit this later when working on improving the Kinto.elm lib, thanks for the feedback ;)